src/operator/controllers/iam/webhooks/pod_webhook.go

```diff
@@ -69,6 +69,7 @@ func (w *ServiceAccountAnnotatingPodWebhook) handleOnce(ctx context.Context, pod
 
 	if !dryRun {
 		apiutils.AddLabel(updatedServiceAccount, w.agent.ServiceAccountLabel(), metadata.OtterizeServiceAccountHasPodsValue)
+		controllerutil.AddFinalizer(updatedServiceAccount, w.agent.FinalizerName())
```

Comment:
Where do we remove this finalizer, once cleanup has completed successfully ? I didn't find the relevant part in the code.

Reply:
once cleanup has completed successfully, here: https://github.com/otterize/credentials-operator/pull/160/files#diff-243aa5a0c536dc773b4ab4dbc8cceef414f7db754b16798256a640a560d0f1caR100